agents/src/transcription.ts

```diff
@@ -35,6 +36,10 @@ export class BasicTranscriptionForwarder implements TranscriptionForwarder {
     this.#messageId = messageId;
   }
 
+  get text(): string {
+    return this.#currentText;
```

Comment:
`#currentText` may have been a poor choice of name because I think (unless I misunderstand this detail of this PR) that you expect it to be only the currently cursored chunk of text? it gets filled in by the realtime api as text is generated, but this happens quickly. then it has a second cursor that we maintain that syncs to playout.

I think you intend to use `.text` here as meaning "the text that has played". if so, then you should use `this.#currentText.slice(0, this.currentCharacterIndex)`

Reply:
that actually breaks things: sometimes the cursor is too early and the end of the transcription gets cut out. `this.#currentText` works great without it